README.md

```diff
@@ -81,6 +81,17 @@ along with the SHA-224 hash of this version for validation.
 recommended as a security practice. Permitted values for the package manager are
 `yarn`, `npm`, and `pnpm`.
 
+You can also provide a URL to a `.js` file (which will be interpreted as a
+CommonJS module) or a `.tgz` file (which will be interpreted as a package, and
+the `"bin"` field of the `package.json` will be used to determine which file to
+use in the archive).
+
+```json
+{
+  "packageManager": "yarn@https://registry.npmjs.org/@yarnpkg/cli-dist/-/cli-dist-3.2.3.tgz#sha224.16a0797d1710d1fb7ec40ab5c3801b68370a612a9b66ba117ad9924b"
```

Comment:
Is the `@` symbol going to conflict with urls that include a username/password? For example, this is a valid url:

```
https://user:pass@example.com
```

Reply:
It shouldn't conflict, we should just split at the first `@` (that's needed anyway to support `@` when used anywhere else in the URL, like here in the scope parameter).

Reply:
`@` is not a valid character in a URL scheme (https://url.spec.whatwg.org/#scheme-state), so I don't think it's going to be a problem. I should add a test that contains `user:pass@example.com` to make sure we support that.

Reply:
Ah good point - there will always be the `@`.

The first part is always the name and the second part is either version or url 👍

```js
const [pkgName, versionOrUrl] = packageManager.split('@')
```

Reply:
> ```js
> const [pkgName, versionOrUrl] = packageManager.split('@')
> ```

Note that if we were using `.split`, it would fail if the URL did contain a `@` char (e.g. `yarn@http://user:pass@example.com`). But yeah, that's the idea.

Reply:
Actually since we landed https://github.com/nodejs/corepack/pull/365, we no longer support URLs with basic auth. I don't think it's a big deal, but if it is, we can add it back later.